### PR TITLE
Contains implementation for feature request PhotoUpload API #13

### PR DESCRIPTION
**Pull request by kspviswa**
I had implemented a new Photo_upload handler to upload photos to pf as a separate api.

HTTP URI : <PF URL>/<REPO>/api/photo_upload?key=<Write API Key>
Eg : http://localhost:8000/haiti/api/photo_upload?key=gD1q-zIfaAo3YoGT

Response will be like this.

```
kspviswa@kspviswa-VirtualBox:/media/sf_Ubuntu_Workspace/kspviswa/pf-viswa$ ./fire.sh 
<?xml version="1.0"?>
<response><url>http://localhost:8000/haiti/photo?id=4642138092470272</url></response>
```

Screen Shots
-----------------
Firing CURL query and getting the Photo-ID
====
image

Querying the photo with generated Photo-id 
=====
image

### CODE REVIEW

**Comment by gimite** [comment]
Thanks for the pull request! I will take a look.

File: app/api.py [hunk 1]
```diff
@@ -180,9 +184,9 @@ def post(self):
             self.error(400, message='Problem in the uploaded file: %s' % e)
         except runtime.DeadlineExceededError, e:
             self.error(400, message=
-                'Sorry, the uploaded file is too large. Try splitting it into '
-                'smaller files (keeping the header rows in each file) and '
-                'uploading each part separately.')
+            'Sorry, the uploaded file is too large. Try splitting it into '
```
Comment (gimite) [inline app/api.py]: It looks you broke indents here and other places. Can you revert them?

File: app/api.py [hunk 2]
```diff
@@ -297,6 +301,37 @@ def get(self):
             self, ApiActionLog.READ, len(records), len(notes))
 
 
+class PhotoUpload(utils.BaseHandler):
+    https_required = True
+
+    def post(self):
+        if not (self.auth):
```
Comment (gimite) [inline app/api.py]: Redundant bracket. Should be if not self.auth:

Comment (gimite) [inline app/api.py]: I believe we should allow this API only for people with "write" access. Can you require self.auth and self.auth.domain_write_permission, in the same way as Write class?

File: app/api.py [hunk 3]
```diff
@@ -297,6 +301,37 @@ def get(self):
...
+        if photo is not None:
+            photo = photo.read()
+            photo = images.Image(photo)
+            photo, photo_url = create_photo(photo, self)
+            if photo:
+                photo.put()
+        else :
+            self.info(
+                403,
+                message='Photo URL is wrong',
```
Comment (gimite) [inline app/api.py]: The message seems wrong. Something like "Request body must not be empty"?

File: app/api.py [hunk 4]
```diff
@@ -297,6 +301,37 @@ def get(self):
...
+    def post(self):
+        if not (self.auth):
+            self.info(
+                403,
+                message='Missing or invalid authorization key',
+                style='plain')
+            return
+        photo = self.request.body_file
+        if photo is not None:
+            photo = photo.read()
```
Comment (gimite) [inline app/api.py]: Can you apply some size restriction (maybe 10MB?) to avoid uploading very large file?

Comment (gimite) [inline app/api.py]: Using the same variable "photo" for various types (file, string, Image) is confusing. Can you use different variable names for each of them?

File: app/api.py [hunk 5]
```diff
@@ -297,6 +301,37 @@ def get(self):
...
+            photo = images.Image(photo)
+            photo, photo_url = create_photo(photo, self)
```
Comment (gimite) [inline app/api.py]: This function can throw SizeTooLargeError and PhotoError. Can you catch the exceptions and output appropriate error message?

File: app/api.py [hunk 6]
```diff
@@ -297,6 +301,37 @@ def get(self):
...
+            if photo:
```
Comment (gimite) [inline app/api.py]: I believe photo can never be None, so this if-statement is not needed.

File: app/api.py [hunk 7]
```diff
@@ -297,6 +301,37 @@ def get(self):
...
+            self.info(
+                403,
+                message='Photo URL is wrong',
+                style='plain')
+            return
+
+        self.response.headers['Content-Type'] = 'application/xml'
+        self.write('<?xml version="1.0"?>\n')
+        self.write('<response><url>')
+        self.write(photo_url)
```
Comment (gimite) [inline app/api.py]: Can you apply XML escaping just in case? It may contain "&" which should be escaped.

**Comment by gimite** [comment]
Overall looks good, thank you! I added some line comments. Can you resolve them?

**Comment by kspviswa** [comment]
@gimite 

Thanks for the review. I shall work on this items and revert back with another commit.

**Comment by kspviswa** [comment]
This pull request is depreciated by https://github.com/google/personfinder/pull/129

**Pull request closed without merging**